### PR TITLE
Issue 959 - Message Handler Service

### PR DESCRIPTION
**Pull request by dfaller**
Created a message handler service that launches long running message handlers to process the backend messages.

1. Created the concept of system services that launch long running system tasks
2. Created the message handler service and added ability to scheduler REST API to control the number of message handlers
3. Added node tasks and system tasks to status REST API
4. Added services section to status REST API

### CODE REVIEW

File: scale/job/tasks/manager.py [hunk 1]
```diff
@@ -20,6 +22,50 @@ def __init__(self):
         self._tasks = {}  # {Task ID: Task}
         self._lock = threading.Lock()
 
+    def generate_status_json(self, nodes_list):
```
Comment (gisjedi) [inline scale/job/tasks/manager.py]: Didn't see any unit test coverage for this function.

File: scale/scheduler/tasks/services/messaging/messaging_service.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+from scheduler.manager import scheduler_mgr
+from scheduler.tasks.services.messaging.message_handler_task import MessageHandlerTask
+from scheduler.tasks.services.service import Service
+
+
+logger = logging.getLogger(__name__)
+
+
+class MessagingService(Service):
```
Comment (gisjedi) [inline scale/scheduler/tasks/services/messaging/messaging_service.py]: Didn't see any unit test coverage for this class.

File: scale/scheduler/tasks/services/service.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import logging
+from abc import ABCMeta, abstractmethod
+
+from job.tasks.update import TaskStatusUpdate
+
+
+logger = logging.getLogger(__name__)
+
+
+class Service(object):
```
Comment (gisjedi) [inline scale/scheduler/tasks/services/service.py]: Didn't see any unit test coverage for this class.

File: scale/scheduler/tasks/system_task.py [hunk 4]
```diff
@@ -46,3 +51,18 @@ def _add_database_docker_params(self):
                      DockerParameter('env', 'SCALE_DB_PORT=%s' % db['PORT'])]
 
         self._docker_params.extend(db_params)
+
+    def _add_messaging_docker_params(self):
```
Comment (gisjedi) [inline scale/scheduler/tasks/system_task.py]: Didn't see any unit test coverage for this function.

File: scale/job/tasks/manager.py (hunk 1 above)
Comment (dfaller) [reply scale/job/tasks/manager.py]: Added unit test

File: scale/scheduler/tasks/services/messaging/messaging_service.py (hunk 2 above)
Comment (dfaller) [reply scale/scheduler/tasks/services/messaging/messaging_service.py]: Unit tests for this class were added

File: scale/scheduler/tasks/services/service.py (hunk 3 above)
Comment (dfaller) [reply scale/scheduler/tasks/services/service.py]: This is an abstract base class. It's concrete methods are tested in the unit tests for the messaging service.

File: scale/scheduler/tasks/system_task.py (hunk 4 above)
Comment (dfaller) [reply scale/scheduler/tasks/system_task.py]: This method is tested within the unit tests for the messaging service. Being a very light method, I don't think a separate test for it is necessary.

**Comment by dfaller** [comment]
I added needed unit tests, so I think we're good to go.

**Pull request merged**